### PR TITLE
Add payment frequency option

### PR DESCRIPTION
**Pull request by lucekdudek**

### CODE REVIEW

**Comment by lucekdudek** [comment]
Requires https://github.com/golemfactory/golem-core-python/pull/131 and golem-core version bump

File: pyproject.toml [hunk 1]
```diff
@@ -10,8 +10,8 @@ packages = [{include = "ray_on_golem"}]
 python = "^3.8"
 ray = {version="^2.7.1", extras=["default"]}
 #golem-core = { path="../golem-core-python", develop=true }
-#golem-core = { git="https://github.com/golemfactory/golem-core-python.git", branch="main" }
-golem-core = "^0.3.2"
+golem-core = { git="https://github.com/golemfactory/golem-core-python.git", branch="main" }
```
Comment (shadeofblue) [inline pyproject.toml:13]: are we sure we want to merge it into `develop` this way?

File: ray_on_golem/server/services/golem/golem.py [hunk 2]
```diff
@@ -153,7 +195,8 @@ async def _create_stack(
                     update_interval=timedelta(seconds=10),
                 ),
                 NegotiatingPlugin(
-                    proposal_negotiators=(AddChosenPaymentPlatform(),),
+                    proposal_negotiators=proposal_negotiators,
+                    proposal_response_timeout=timedelta(seconds=30),
```
Comment (shadeofblue) [inline ray_on_golem/server/services/golem/golem.py:199]: I'd also turn this one into a constant defined above

**Review by shadeofblue — APPROVED** [review]
please see my comments

File: pyproject.toml (hunk 1 above)
Comment (lucekdudek) [reply pyproject.toml:13]: I think we are

File: ray_on_golem/server/models.py [hunk 3]
```diff
@@ -66,13 +66,27 @@ class PerCpuExpectedUsageData(BaseModel):
     max_cost: Optional[float] = None
 
 
+class PaymentFrequency(BaseModel):
+    minimal: int
+    optimal: int = None
```
[the remark this answers is not in the document]
Comment (lucekdudek) [reply ray_on_golem/server/models.py:71]: Float should work too actually

File: ray_on_golem/server/services/golem/golem.py [hunk 4]
```diff
@@ -120,18 +128,53 @@ async def _create_stack(
...
+            demand_lifetime = DEFAULT_LONG_RUNNING_DEMAND_LIFETIME
+
+            proposal_negotiators.append(
+                MidAgreementPaymentsNegotiator(
+                    min_debit_note_interval=DEFAULT_DEBIT_NOTE_INTERVAL,
+                    optimal_debit_note_interval=DEFAULT_DEBIT_NOTE_INTERVAL,
+                    min_payment_timeout=minimal_payment_timeout,
+                    optimal_payment_timeout=optimal_payment_timeout,
+                )
+            )
```
[the remark this answers is not in the document]
Comment (approxit) [reply ray_on_golem/server/services/golem/golem.py]: For me it looks like `extra_proposal_plugins` and `extra_proposal_scorers`. Currently it only exists for the sake of `ManagerStackNodeConfigHelper`. I see `extra_proposal_negotiators` as something that should be added alongside moving highlighted chunk of code to separate method.

**Pull request merged**